src/main/java/uk/gov/hmcts/reform/sandl/snlevents/service/HearingPartService.java

```diff
@@ -95,19 +97,21 @@ public UserTransaction assignWithTransaction(Hearing hearing, UUID transactionId
                 0)
         );
 
-        HearingPart hearingPart = hearing.getHearingParts().get(0);
-        userTransactionDataList.add(new UserTransactionData("hearingPart",
-            hearingPart.getId(),
-            beforeHearingPart,
-            "update",
-            "update",
-            1));
+        AtomicInteger index = new AtomicInteger();
 
+        hearing.getHearingParts().forEach(hp -> {
+            userTransactionDataList.add(new UserTransactionData("hearingPart",
+                hp.getId(),
+                previousHearingParts.get(index.getAndIncrement()),
+                "update",
+                "update",
+                1
```

Comment:
so if there will be multiple hearing parts all of them will have order 1, will that not break later actions?

Reply:
No, it won't.

Reply:
What actions do you mean and why would it?

Reply:
in this case not as these are the same type of actions the order matter i.e. when you first need to modify hearingpart and then session not the other way round because it would crash. here the hearingparts do not depend one on another.